android/sdl_android/src/main/java/com/smartdevicelink/transport/RouterServiceValidator.java

```diff
@@ -200,6 +215,187 @@ public boolean validate(){
 		return false;
 	}
 
+	/**
+	 * Asynchronously validate the target RouterService, which includes finding the right RouterService.
+	 * @param callback: callback gets called when validation finishes.
+	 */
+	public void validateAsync(final ValidationStatusCallback callback) {
+		if(securityLevel == -1){
+			securityLevel = getSecurityLevel(context);
+		}
+
+		final PackageManager pm = context.getPackageManager();
+		//Grab the package for the currently running router service. We need this call regardless of if we are in debug mode or not.
+
+		if(this.service != null){
+			Log.i(TAG, "Supplied service name of " + this.service.getClassName());
```

Comment:
Can we also use the `DebugTool` for this log and the one on line 235?

Reply:
Well, I was just copied from validate() method, but anyway, I will update this to DebugTool as well as Line 235.